RadarSDK/RadarLog.m

```diff
@@ -0,0 +1,53 @@
+//
+//  RadarLog.m
+//  RadarSDK
+//
+//  Copyright © 2021 Radar Labs, Inc. All rights reserved.
+//
+
+#import "RadarLog.h"
+
+@implementation RadarLog
+
+- (instancetype _Nullable)initWithMessage:(NSString *_Nullable)message
+                                    level:(RadarLogLevel)level {
+    self = [super init];
+    if (self) {
+        _level = level;
+        _message = message;
+        _createdAt = [NSDate new];
+    }
+    return self;
+}
+
+- (NSDictionary *)dictionaryValue {
+    NSMutableDictionary *dict = [NSMutableDictionary new];
+    dict[@"level"] = [Radar stringForLogLevel:self.level];
+    dict[@"message"] = self.message;
+
+    // convert (NSDate)createdAt to ISO8601 string
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    dateFormatter.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_US_POSIX"];
+    dateFormatter.timeZone = [NSTimeZone timeZoneWithAbbreviation:@"UTC"];
+    [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSSZ"];
```

Comment:
1. `NSDateFormatter`s should be cached because they don't change, and they can be expensive to create.
2. There's an `NSISO8601DateFormatter` class that should do this out of the box, or with some options.

Reply:
+1 to the above - let's cache the `dateFormatter` and not recreate it every time this fn is called

Reply:
Looks like the exact same format string is used twice in `RadarEvent.m` and once in `RadarUserInsightsLocation.m`.

Reply:
@corypisano, I can make that change and check it in on this branch, if you want.

Reply:
sure!

Reply:
Done.

Reply:
awesome thanks 👍